internal/service/encryptionatrest/data_source_schema.go

```diff
@@ -0,0 +1,184 @@
+package encryptionatrest
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func DataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"aws_kms_config": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"access_key_id": schema.StringAttribute{
+						Computed:            true,
+						Sensitive:           true,
+						Description:         "Unique alphanumeric string that identifies an Identity and Access Management (IAM) access key with permissions required to access your Amazon Web Services (AWS) Customer Master Key (CMK).",
+						MarkdownDescription: "Unique alphanumeric string that identifies an Identity and Access Management (IAM) access key with permissions required to access your Amazon Web Services (AWS) Customer Master Key (CMK).",
+					},
+					"customer_master_key_id": schema.StringAttribute{
+						Computed:            true,
+						Sensitive:           true,
+						Description:         "Unique alphanumeric string that identifies the Amazon Web Services (AWS) Customer Master Key (CMK) you used to encrypt and decrypt the MongoDB master keys.",
+						MarkdownDescription: "Unique alphanumeric string that identifies the Amazon Web Services (AWS) Customer Master Key (CMK) you used to encrypt and decrypt the MongoDB master keys.",
+					},
+					"enabled": schema.BoolAttribute{
+						Computed:            true,
+						Description:         "Flag that indicates whether someone enabled encryption at rest for the specified project through Amazon Web Services (AWS) Key Management Service (KMS). To disable encryption at rest using customer key management and remove the configuration details, pass only this parameter with a value of `false`.",
+						MarkdownDescription: "Flag that indicates whether someone enabled encryption at rest for the specified project through Amazon Web Services (AWS) Key Management Service (KMS). To disable encryption at rest using customer key management and remove the configuration details, pass only this parameter with a value of `false`.",
```

Comment:
As shared in the slack thread since this is being fetched directly from the API spec, would like to keep these in alignment, have applied the necessary comments though, please lmk if you still have any concerns:)